### DISCUSSION TITLE
Create job definitions from StaticPartitionsDefinition

### DISCUSSION THREAD

**Discussion by AlejandroUPC** [dagster-io/dagster#16987] [Q&A]
Based on this example [12061](https://github.com/dagster-io/dagster/discussions/12061) from @sryza.

Mmm I am struggling here to schedule a job on a fixed frequency for those assets based on a static partition even after reading here https://discuss.dagster.io/t/2645595/hello-i-have-another-static-partition-question-i-have-a-bunc

The following code (adapted to your example) trigers this error:

```python
@repository
def repo():
    return [customers_partitions_def, define_asset_job("my_job", partitions_def=customers_partitions_def )]

```
>  UserWarning: Error loading repository location repository.py:dagster._core.errors.DagsterInvalidDefinitionError: Bad return value from repository construction function: all elements of list must be of type JobDefinition, GraphDefinition, ScheduleDefinition, SensorDefinition, AssetsDefinition, SourceAsset, or AssetChecksDefinition.Got value of type <class 'dagster._core.definitions.partition.StaticPartitionsDefinition'> at index 0.

I am not sure how to convert the StaticPartitionsDefinition to the proposed types?

And removing it loads the job but without any op to run:

```python

@repository
def repo():
    return [define_asset_job("my_job", partitions_def=customers_partitions_def )]
```

I would like to be able to schedule a job that executes under a given cron frequency for a group of assets that have been defiend using `StaticPartitionsDefinition`

**Comment by sryza** [comment]
Hi @AlejandroUPC - this line is including the PartitionsDefinition directly on the repository:
```python
    return [customers_partitions_def, define_asset_job("my_job", partitions_def=customers_partitions_def )]
```

Instead of `customers_partitions_def`, you should include the asset definition.

**Reply by AlejandroUPC** [reply]
Mm I am not sure how can I get an asset definition from a pipeline of asstes where asset2 depends on asset1?

**Reply by sryza** [reply]
You'd want to include the asset objects:

```python

@asset(partitions_def_...)
def asset1():
    ...

@asset(partitions_def_...)
def asset2():
    ...

@repository
def repo():
    return [asset1, asset2, define_asset_job(...)]
```
